[New Tutorial] building-multi-region-disaster-recovery-for-amazon-appstream2.0 (#636)
build-on-aws/content: tutorials/building-multi-region-disaster-recovery-for-amazon-appstream2.0/index.md

---
title: "Building a Multi-Region Disaster Recovery Environment for Amazon AppStream 2.0"

> description: "A step by step guide on how to build a multi-Region disaster recovery environment for Amazon AppStream 2.0 using FSLogix CloudCache user profile containerization and replication software."

Reviewer: ```suggestion
Trimmed to accomodate less characters for easier "hook":
A guide on FSLogix CloudCache and replication software for resilience. 

```

tags:
  - aws
  - disaster-recovery
  - appstream
  - end-user-computing
  - resilience

authorGithubAlias: Peter-John-M
authorName: Peter-John Morgenrood
date: 2023-08-16

> ---

Reviewer: Hello! This is a very important topic. It's got good structure and easy to follow steps. I made some suggestions on breaking up the intro. I corrected titles to match AP Style and made a couple of tweaks. Overall, it's pretty solid and didn't need much tinkering. 

Good job! 👏

Author: Thank you very much! Appreciate your review, and I have completed all suggested revisions. 
I showed the architecture diagram to a friend who has an AWS cloud practitioner certification but does not work in AWS daily. 
He suggested I add some arrows and labels to make things more clear to a non AWS\wider audience. (done)
All is looking good now, looking forward to final review.

Reviewer: This is solid! The only thing is that I didn't see the titles updated to AP Style. That's part of Amazon's publishing guide.  Once those updates are done, this will be ready to publish.

Author: Done!

3.  Add some files to your Documents folder

![EC2 Windows File Explorer Window, Adding test data called TPS Report Documents to the user profile, Documents folder in order to simulate profile size increase.](images/image11.png "Figure 11. EC2 Windows File Explorer Adding Documents to user profile")

4.  Verify profiles sizes have increased in both Primary and DR SMB
    locations

![Windows file explorer app with two windows, one connected to the primary file server and the other connected to the DR file server. Primary and DR file server windows both have a file called profile_as2test2.vhdx with a new size of 397,312 KB](images/image12.png "Figure 12. EC2 Windows File Explorer Connected to FSx Primary and DR" )

> 5.  End the Amazon AppStream 2.0 streaming session

Reviewer: ```suggestion
5.  End the Amazon AppStream 2.0 streaming session.
```

6.  Simulate a region failure by blocking all inbound TCP port 445 to
    Primary SMB location.

> 7.  Connect to your Okta DR application, in my example it the Okta

Reviewer: ```suggestion
7.  Connect to your Okta DR application, in my example the Okta
```

>     application would be called “AppStream 2.0 DR”

Reviewer: ```suggestion
    application would be called “AppStream 2.0 DR”.
```

8.  If you have logged in and your documents are all there, it means at
    this point you have successfully failed over and working in the DR

>     region

Reviewer: ```suggestion
    region.
```

9.  Add some more test files to your Documents folder on the DR fleet
    which will automatically get synced back once the Primary file
    server port 445 is reachable from the DR fleet.

10. Enable Inbound TCP port 445 for Primary file server SMB location and
    verify that all the newly added files from DR fleet have synced back

>     to your primary region

Reviewer: ```suggestion
    to your primary region.
```

11. After testing, you can stop the fleet in your DR region to save on
    running costs.

> ### Clean-up:

Reviewer: Clean-Up

In this blog, we deployed resources in an AWS account and made
configuration changes in Active Directory. If you want to clean up these

> resources and reverse changes you made, you can complete the following

Reviewer: ```suggestion
resources and reverse changes you made, complete the following
```

I encourage you to explore these concepts further and think holistically about how common settings like Outlook [Cached Exchange Mode](https://support.microsoft.com/en-us/office/turn-on-cached-exchange-mode-7885af08-9a60-4ec3-850a-e221c1ed0c1c), can be configured to reduce your cloud storage costs by limiting the past email to a few months instead of 1 year. 

This tutorial sets the foundations that enables you to use your existing FSLogix configuration to explore and deploy advanced controls like [Application Masking](https://aws.amazon.com/blogs/desktop-and-application-streaming/simplify-appstream-2-0-image-management-with-application-masking?sc_channel=el&sc_campaign=resiliencewave&sc_geo=mult&sc_country=mult&sc_outcome=acq&sc_content=mr-dr-for-appstream), to limit apps based on Active Directory group membership. This is great for users sharing a single Amazon AppStream fleet. 

> 

Reviewer: Consider adding a link or two to other blogs that speak to this topic.
EX: "Learn more about XYZ..."

---
title: "Building a multi-region disaster recovery environment for Amazon AppStream 2.0"

> description: "A step by step guide on how to build multi region disaster recovery environment for Amazon AppStream 2.0 using FSLogix CloudCache a user profile containerization and replication software."

Reviewer 2: multi-region (like you did in the title).

Note that AWS convention is to capitalize "Region", so multi-Region, but it is optional if you want to follow that here

Author: I have made the title all caps on the first letter so that Region does not stand out too much.
Current:
"Building a Multi-Region Disaster Recovery Environment for Amazon AppStream 2.0"

Compared to:
"Building a multi-region disaster recovery environment for Amazon AppStream 2.0"

Added the capital R in multi-Region as per the AWS convention.
description: "A step by step guide on how to build a multi-Region disaster recovery environment for Amazon AppStream 2.0 using FSLogix CloudCache user profile containerization and replication software."

> tags:

Reviewer 2: add "resilience"

  - aws

>   - disaster recovery

Reviewer 2: disaster-recovery

>   - appstream 2.0

Reviewer 2: this would be a new tag, but no spaces or dots allowed, would just "appstream" be ok?

Author: Perfect, appstream can work.

>   - end user computing

Reviewer 2: end-user-computing

Author: Done

> spaces:

Reviewer 2: delete the spaces and its value -- we do no have a space defined for end-user-computing.  We may in the future.  I may add this to the resilience space, but that is not yet configured

  - end user computing
authorGithubAlias: Peter-John-M
authorName: Peter-John Morgenrood
date: 2023-08-16
---

> 

Reviewer 2: add `## Introduction`

Author: Added ## Introduction on line 17 after date.

region during the outage and in some cases, lose data or access to user profile settings.
Rebuilding can be difficult and time consuming, which can impact revenue. There is potential for human error when rebuilding a new environment under production downtime pressure. Planning and maintaining
a disaster recovery mechanism to avoid compromising business performance is essential for businesses of all sizes.

AWS has published [Disaster Recovery considerations with Amazon
AppStream
2.0](https://aws.amazon.com/blogs/desktop-and-application-streaming/disaster-recovery-considerations-with-amazon-appstream-2-0/)
and [Optimize User Experience with latency-based routing for Amazon
AppStream
2.0](https://aws.amazon.com/blogs/desktop-and-application-streaming/optimize-user-experience-with-latency-based-routing-for-amazon-appstream-2-0/)
that address DR for customer environments that are using [Home Folder
Synchronization](https://docs.aws.amazon.com/appstream2/latest/developerguide/home-folders.html)<u>,</u>
or have enabled [Application Settings
Persistence](https://docs.aws.amazon.com/appstream2/latest/developerguide/app-settings-persistence.html)

> While these are excellent solutions, there are customers that require the user profile sizes to be larger than the 1GB limit of Amazon AppStream 2.0 App Settings Persistence feature and be able to dynamically expand profile size on demand with limited impact to log-on times.

Reviewer 2: > While these are excellent solutions, there are customers that require t

Can give more detail why the above solutions do not work in these cases?

1. Replication of user profile begins when the user logs on to their virtual desktop and Cloud Cache copies the incremental changes to DR location automatically.
1. AppStream 2.0 users can share folders between different AppStream fleets, which was challenging when using S3 for home folders. 
1. Terabytes of user profile and other data stored on premise can be accessed securely and remotely using AppStream 2.0 and [Amazon FSx File Gateway](https://aws.amazon.com/storagegateway/file/fsx/), which provide seamless read and write activity when files are shared between their on-premises locations and the cloud. Users will have a similar experience inside a virtual desktop as they do on-premise using a physical laptop\device
1. Fine grained access control of data from a central source such as Active Directory (AD) that integrates with existing onboarding workflows and security folder monitoring tools.

In this blog, I will guide you through how to build a multi-region disaster recovery environment for Amazon AppStream 2.0 using Amazon FSx for Windows as a storage location.

> ### Prerequisites

Reviewer 2: I think this should be `##` 

`##` is your top-level header (you should not use `#`)

    System](https://docs.aws.amazon.com/fsx/latest/WindowsGuide/what-is.html)
    used for profile storage, check [this blog
    post](https://aws.amazon.com/blogs/desktop-and-application-streaming/use-amazon-fsx-and-fslogix-to-optimize-application-settings-persistence-on-amazon-appstream-2-0/)
    for setup instructions

6.  The VPC subnets for Amazon AppStream 2.0 in primary and disaster
    recovery region, must be able to connect on port 445, to both
    primary and disaster recovery FSx for Windows File Systems.

7.  Both VPCs connected via Transit Gateway or VPC Peering  
    ([VPC peering is preferable in most
    settings](https://docs.aws.amazon.com/whitepapers/latest/best-practices-for-deploying-amazon-appstream-2/availability-zones.html))

8.  A SAML IdP such, deployed in two or more regions like
    [Okta](https://www.okta.com/sites/default/files/2020-10/Okta%27s-Architecture-eBook.pdf)
    which is able to endure a region failure.

**Note**

*AWS IAM Identity Center (Successor to AWS Single Sign-On) does not
support multi-region at this time and can only be deployed in one
region, per AWS Organization.*

# Solution overview

> Figure 1 shows the components and traffic flow for the solution.

Reviewer 2: For all figures and captions you can do this instead

To show images in your post, you'll link to them using a relative path:

```
![Put your alt-text here](images/the-internet.jpg "Put your image title/caption here")
```

The caption will appear underneath the image and also on mouse over.

This and other shortcuts can be found here: https://github.com/build-on-aws/content/blob/bef4d1843d0eb67981cfb0e5e5ff6af8a106b679/FAQ.md

Author: Mouse over and alt text is working now, thank you for catching this.

<img src="images/image1.png" style="width:6.5in;height:3.69136in" />

Figure 1: Solution architecture

The high level steps is as follows:

1.  Setup file share and folder permissions to store user profile data

2.  Setup share permissions to allow data write and replication over
    network

3.  Install and configure FSLogix on the Amazon AppStream 2.0
    ImageBuilder

4.  Setup the Group Policy to configure FSLogix

5.  Create an Amazon AppStream 2.0 image

6.  Copy the Amazon AppStream 2.0 image to the DR region

7.  Recreate the Amazon AppStream 2.0 Stack configuration in the DR
    region

8.  Setup your SAML IdP

9.  Test if Disaster Recovery is working as expected

> # Deploy the solution

Reviewer 2: make this H2 `##`, and adjust the headers under it

Author: Added ## to Deploy the solution and changed all heading below to ### .

> ## Step 1: Setup file and folder permissions

Reviewer 2: all steps should be `###`

Author: Done

2.  Once connected to the fleet, use file explorer to connect to your
    SMB locations 
    ```
    \\FSxPrimaryDNSName\Profiles
    ```
    and confirm that the user profile folder and vhdx file appears under the specified SMB    locations, set in step 3. Take note of the initial profile size

<img src="images/image10.png" style="width:7.77657in;height:6.26411in"
alt="Windows file explorer app with two windows, one connected to the primary file server and the other connected to the DR file server. Primary and DR file server windows both have a file called profile_as2test2.vhdx with a size of 167,936 KB Primary and DR have a file called Profile_as2test2.VHDX.lock 3KB in size but Primary Profile_as2test2.VHDX.meta is 1KB While DR Profile_as2test2.VHDX.meta is 0KB " />

*Figure 10: EC2 Windows File Explorer Connected to FSx Primary and DR*

3.  Add some files to your Documents folder

<img src="images/image11.png" style="width:9.01366in;height:4.39416in"
alt="EC2 Windows File Explorer Window, Adding test data called TPS Report Documents to the user profile, Documents folder in order to simulate profile size increase." />

> *Figure 11: EC2 Windows File Explorer Adding Documents to user profile*

Reviewer 2: is showing the desktop an artistic choice?  Can we trim to just the window for all of these?  your call

Author: Yes, you are correct for image 10 and 12, I can trim them out but image 11 (this image) I would like to keep it for context because it shows the tool bar of AppStream instance at the top. Image 10 and 12, I will go ahead and trim it out, as its not needed here.

Author: I trimmed image 11 to remove the desktop but but kept the AppStream toolbar at the top.

    server port 445 is reachable from the DR fleet.

10. Enable Inbound TCP port 445 for Primary file server SMB location and
    verify that all the newly added files from DR fleet have synced back
    to your primary region

11. After testing, you can stop the fleet in your DR region to save on
    running costs.

## Clean-up:

In this blog, we deployed resources in an AWS account and made
configuration changes in Active Directory. If you want to clean up these
resources and reverse changes you made, you can complete the following
steps:

1.  Unlink and delete the FSLogix GPO you created in the section **Setup
    Group Policy to** **Configure FSLogix**. If you want, you can remove
    the FSLogix Administrative templates you deployed to your central
    store.

2.  Remove users from your AppStream 2.0/FSLogix Active Directory group.

3.  Delete your Amazon FSx for Windows File Server file systems in both
    regions.

4.  Delete or stop any AppStream 2.0 fleets, images, stacks, and image
    builders you have created specifically for this blog.

5.  Delete the IAM Roles and SAML Identity providers

6.  Delete the Okta Applications created for this blog

> # Conclusion

Reviewer 2: `##`

Author: Added ## to conclusion.

---

> title: "Building a multi-region disaster recovery environment for Amazon AppStream 2.0"

Reviewer: ```suggestion
title: "Building a Multi-Region Disaster Recovery Environment for Amazon AppStream 2.0"
```
